risc0/cargo-risczero/src/commands/datasheet/mod.rs

```diff
@@ -0,0 +1,489 @@
+    }
+
+    pub fn execute_with_segments<F>(
+        client: &ApiClient,
+        env: &ExecutorEnv,
+        elf: &[u8],
+        segment_callback: F,
+    ) -> Result<SessionInfo>
+    where
+        F: FnMut(SegmentInfo, Asset) -> Result<()>,
+    {
+        let binary = Asset::Inline(elf.to_vec().into());
+        let segments_out = AssetRequest::Inline;
+        client.execute(env, binary, segments_out, segment_callback)
+    }
+
+    pub fn prove(
+        client: &ApiClient,
+        env: &ExecutorEnv,
+        elf: &[u8],
+        opts: &ProverOpts,
+    ) -> Result<ProveInfo> {
+        let image_id = risc0_zkvm::compute_image_id(elf)?;
+        let binary = risc0_zkvm::Asset::Inline(elf.to_vec().into());
+        let prove_info = client.prove(env, opts, binary)?;
+
+        let ctx = VerifierContext::default();
+        if opts.prove_guest_errors {
+            prove_info.receipt.verify_integrity_with_context(&ctx)?;
+        } else {
+            prove_info.receipt.verify_with_context(&ctx, image_id)?;
+        }
+
+        Ok(prove_info)
+    }
+
+    pub fn loop_env<'a>(iters: u32) -> Result<ExecutorEnv<'a>> {
+        loop_env_builder(iters).build()
+    }
+
+    pub fn loop_env_builder<'a>(iters: u32) -> ExecutorEnvBuilder<'a> {
+        let mut builder = ExecutorEnvBuilder::default();
+        builder.write_slice(&iters.to_le_bytes());
+        builder
+    }
+}
+
+/// Benchmark implementations.
+mod benches {
+    use risc0_zkvm::InnerReceipt;
+
+    use super::*;
+
+    const RECURSION_CYCLES: u64 = 1 << RECURSION_PO2;
+
+    pub fn execute(client: &ApiClient, iters: u32) -> Result<BenchmarkData> {
+        println!("execute: {iters}");
+
+        let env = util::loop_env(iters)?;
+        let (session, duration) = try_time(|| util::execute(client, &env, LOOP_ELF))?;
+
+        // NOTE: We use user cycles as the total because there is no proving.
+        let cycles = session.cycles();
+
+        Ok(BenchmarkData {
+            name: "execute",
+            hashfn: "N/A".into(),
+            throughput: cycles as f64 / duration.as_secs_f64(),
+            duration,
+            cycles,
+            seal: 0,
+        })
+    }
+
+    pub fn composite(
+        client: &ApiClient,
+        hashfn: &str,
+        iters: u32,
+        po2: u32,
+    ) -> Result<BenchmarkData> {
+        let expected_cycles = 1u64 << po2;
+        println!("rv32im ({hashfn}): {expected_cycles}");
+
+        let opts = ProverOpts::default().with_hashfn(hashfn.to_string());
+        let env = util::loop_env(iters)?;
+        let (info, duration) = try_time(|| util::prove(client, &env, LOOP_ELF, &opts))?;
+
+        let cycles = info.stats.total_cycles;
+        assert_eq!(
+            cycles, expected_cycles,
+            "Incorrect cycle count for po2={po2}"
+        );
+
+        Ok(BenchmarkData {
+            // XXX: This differs from the filter. Should we have these match?
+            name: "rv32im",
```

Comment:
@flaub `composite` and `identity_p254` differ between their output names and filter names. Shouldn't these match?

Reply:
what'a a filter name and an output name?

Reply:
`cargo risczero datasheet identity` vs `identity_p254` in the stdout table and JSON file.